File: backend_project_centropsicologico_pisw2025-main/mer/src/main/java/com/sensespsicologos/mer/models/District.java
package com.sensespsicologos.mer.models;

import com.sensespsicologos.mer.common.models.NamedEntity;
import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Entity
@Table(name = "districts")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class District extends NamedEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "province_id", nullable = false)
    private Province province;

    @OneToMany(mappedBy = "district", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Location> locations = new ArrayList<>();

    @OneToMany(mappedBy = "district", cascade = CascadeType.ALL)
    private Set<Patient> patients = new HashSet<>();
}

File: backend_project_centropsicologico_pisw2025-main/src/interfaces/District.ts
import { Location } from "./Location";
import { Patient } from "./Patient";
import { Province } from "./Province";

export interface District {
  id: string;
  name: string;

  provinceId: string;
  province: Province;

  locations: Location[];

  patients: Patient[];

  createdAt: Date;
  updatedAt: Date;
}
